__tests__/app.test.js

```diff
@@ -113,4 +113,57 @@ describe("GET/not-a-route", () => {
           });
         });
       });  
-    });
\ No newline at end of file
+    });
+describe("GET /api/articles/:article_id/comments", () => {
+    it("responds with an array of comments for the given article_id of which each comment should have the correct properties", () => {
+        const articleId = 1
+        return request(app)
+            .get(`/api/articles/${articleId}/comments`)
+            .expect(200)
+            .then((res) => {
+                const { body } = res;
+                const { comments } = body;
+                expect(comments.length).toBeGreaterThan(0);
+                comments.forEach((comment) => {
+                    expect(comment).toHaveProperty("comment_id", expect.any(Number));
+                    expect(comment).toHaveProperty("votes", expect.any(Number));
+                    expect(comment).toHaveProperty("created_at", expect.any(String));
+                    expect(comment).toHaveProperty("author", expect.any(String));
+                    expect(comment).toHaveProperty("body", expect.any(String));
+                    expect(comment).toHaveProperty("article_id", expect.any(Number));
+                });
+            });
+    });
+    it("404: responds with an error if article_id is not found", () => {
+        const articleId = 0;
+        return request(app)
+            .get(`/api/articles/${articleId}`)
+            .expect(404)
+            .then((res) => {
+                const { body } = res;
+                expect(body.msg).toBe(`Article ${articleId} not found`);
+            });
+    });
+    it("400: responds with an error if article_id is not a number", () => {
+        const articleId = 'not_a_number';
+        return request(app)
+            .get(`/api/articles/${articleId}`)
+            .expect(400)
+            .then((res) => {
+                const { body } = res;
+                const { msg } = body;
+                expect(msg).toBe('Invalid article ID');
+            });
+    });
+    it("200: responds with an empty array if article_id is valid but has no comments", () => {
+        const articleId = 2;
+        return request(app)
+            .get(`/api/articles/${articleId}/comments`)
+            .expect(200)
+            .then((res) => {
+                const { body } = res;
+                const { comments } = body;
+                expect(comments.length).toBe(0);
+            });
+    });
+});
```

Comment:
Nice testing so far! Could we just add one more to check reviews are sorted in descending order please?

Reply:
Hey, thanks for your feedback, test added